indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -1118,6 +1141,7 @@ private void taskComplete(
     runningTasks.remove(taskStatus.getId());
 
     // Notify interested parties
+    TaskRunnerUtils.notifyStatusChanged(listeners, taskStatus.getId(), taskStatus);
     taskRunnerWorkItem.setResult(taskStatus);
```

Comment:
Please switch these to keep the setResult -> notifyStatusChanged order consistent every time we do that. I think this matters because setResult is what updates TaskStorage (through a callback) and notifyStatusChanged is what triggers another supervisor run. Since supervisor does a getStatus we want that update to have already happened.